src/components/Panel.ts

```diff
@@ -31,30 +31,39 @@ class Panel implements FlickingPanel {
   private clonedPanels: Panel[];
 
   public constructor(
-    element: HTMLElement,
-    index: number,
-    viewport: Viewport,
+    element?: HTMLElement | null,
+    index?: number,
+    viewport?: Viewport,
   ) {
-    this.viewport = viewport;
+    this.viewport = viewport!;
     this.prevSibling = null;
     this.nextSibling = null;
     this.clonedPanels = [];
 
     this.state = {
-      index,
+      index: index!,
       position: 0,
       relativeAnchorPosition: 0,
       size: 0,
       isClone: false,
       isVirtual: false,
       cloneIndex: -1,
       originalStyle: {
-        className: element.getAttribute("class"),
-        style: element.getAttribute("style"),
+        className: "",
+        style: "",
       },
       cachedBbox: null,
     };
+    this.initElement(element);
+  }
 
+  public initElement(element?: HTMLElement | null) {
+    if (!element) {
+      return;
+    }
+    this.element = element;
```

Comment:
@daybrush setElement does `this.element = element` also.

Reply:
@jongmoon 
setElement is a change in element.